acslib/ccure/crud.py

```diff
@@ -1,28 +1,34 @@
 from typing import Optional
 
-from acslib.base import ACSRequestData, ACSRequestResponse, ACSRequestException, status
-from acslib.base.connection import ACSRequestMethod, ACSConnection
+from acslib.base import ACSRequestData, ACSRequestResponse
+from acslib.base.connection import ACSRequestMethod
 from acslib.ccure.base import CcureACS
 from acslib.ccure.connection import CcureConnection
-from acslib.ccure.search import PersonnelFilter, ClearanceFilter, CredentialFilter
-from acslib.ccure.types import CredentialCreateData
+from acslib.ccure.search_filtering import (
+    PersonnelFilter,
+    ClearanceFilter,
+    CredentialFilter,
+    ClearanceItemFilter
+)
+from acslib.ccure.types import CredentialCreateData, ClearanceItemTypes, ClearanceItemCreateData
 
 
 class CcureAPI:
     def __init__(self, connection: Optional[CcureConnection] = None):
-        self.personnel = CCurePersonnel(connection)
-        self.clearance = CCureClearance(connection)
-        self.credential = CCureCredential(connection)
+        self.personnel = CcurePersonnel(connection)
+        self.clearance = CcureClearance(connection)
+        self.credential = CcureCredential(connection)
+        self.clearance_item = CcureClearanceItem(connection)
 
 
-class CCurePersonnel(CcureACS):
+class CcurePersonnel(CcureACS):
     def __init__(self, connection: Optional[CcureConnection] = None):
         super().__init__(connection)
-        self.search_filter = PersonnelFilter()
+        self.default_search_filter = PersonnelFilter()
```

Comment:
`search_filter` is part of the base class. Why do we need a new attribute?

Reply:
I wasn't thinking about the base class, but it seemed clearer and more readable for the default filter to have a name that clarifies its purpose in the code. 
`search_filter or self.default_search_filter` feels more intuitive to me than `search_filter or self.search_filter`

Reply:
`search_filter` can be both a default (when we the package maintainers set it) and a user defined field. So we are developing with a easy to use and write pattern in mind. However, the design of our package allows the developer to do something more complex like this:

```python
from acslib import CcureAPI
from acslib.ccure.filters import ClearanceItemFilter, NFUZZ, etc

clearance_item = CcureAPI().clearance_item
kwargs = {
 "stuff": "my_stuff"
}
clearance_item.search_filter = ClearanceItemFilter(**kwargs)
clearance_item.search()
....

```

And in this case, or something like it, `default_search_filter` feels not correct to me.

I personally don't have much problem with the `search_filter or self.search_filter` since the scope of the term is clearly defined as `local` and `class`.